### PR TITLE
Stopping Pair<L, R> of implementation package from leaking into public surface area 

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
Stopping Pair<L, R> of implementation package from leaking into public surface area for readMany API

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemIdentity.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.models;
+
+/**
+ * Encapsulates the PartitionKey/id tuple that uniquely identifies a CosmosItem
+ */
+public class CosmosItemIdentity {
```
Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemIdentity.java:9]: @kushagraThapar this is a new public api 
FYI

Comment (moderakh) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemIdentity.java:9]: @FabianMeiswinkel shouldn't this class be marked with BETA annotation?

**Review by moderakh — APPROVED** [review]
One minor comment on adding beta annotation on the `CosmosItemIdentity` class. other than that LGTM.

@kushagraThapar There is a new public class `CosmosItemIdentity` used in the beta api.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemIdentity.java (hunk 1 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemIdentity.java:9]: We should mark this as `final` class.

Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemIdentity.java:9]: We should discuss this. 
@FabianMeiswinkel  @moderakh - We have a release going out early next week, probably 9/8. 
If we can finalize this soon, we will be able to get this in.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ItemOperations.java [hunk 2]
```diff
@@ -53,7 +56,27 @@
             throw new IllegalArgumentException("Parameter container is required and cannot be null.");
         }
 
-        return container.readMany(itemKeyList, classType);
+        return container.readMany(convertItemKeyList(itemKeyList), classType);
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ItemOperations.java:59]: These APIs are also in implementation layer, we have ability to still improve them. Shouldn't we change their signature to take `CosmosItemIdentity` ?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ItemOperations.java [hunk 3]
```diff
@@ -31,6 +33,7 @@
     public static <T> FeedResponse<T> readMany(CosmosContainer container,
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ItemOperations.java:33]: These APIs are also in implementation layer, we have ability to still improve them. Shouldn't we change their signature to take `CosmosItemIdentity` ?

**Review by kushagraThapar — CHANGES_REQUESTED** [review]

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ItemOperations.java (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ItemOperations.java:59]: No - I left this intentionally to not break Adobe. Sounds like we told Adobe to take a dependency on this implementation package class. Whoever worked with Adobe on this needs to follow up with them and ask them to move to the ReadMany Api in the public surface area - only then we will be able to delete this class.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ItemOperations.java (hunk 3 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ItemOperations.java:33]: No - I left this intentionally to not break Adobe. Sounds like we told Adobe to take a dependency on this implementation package class. Whoever worked with Adobe on this needs to follow up with them and ask them to move to the ReadMany Api in the public surface area - only then we will be able to delete this class.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemIdentity.java (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemIdentity.java]: Fixed

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosItemIdentity.java]: Fixed

**Comment by FabianMeiswinkel** [comment]
@kushagraThapar added the Beta-Annotation to CosmosItemIdentity and marked it as final as requested. Not changing ItemOptions was intentional - this class is used by Adobe right now - wanted to avoid breaking them. We need to work with them to move to the new Api in the public surface area and then can delete ItemOperations altogether. Makes sense?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ItemOperations.java (hunk 2 above)
Comment (kushagraThapar) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ItemOperations.java:59]: Sounds good, makes sense.

**Comment by kushagraThapar** [comment]
> @kushagraThapar added the Beta-Annotation to CosmosItemIdentity and marked it as final as requested. Not changing ItemOptions was intentional - this class is used by Adobe right now - wanted to avoid breaking them. We need to work with them to move to the new Api in the public surface area and then can delete ItemOperations altogether. Makes sense?

Yes, that makes sense. Let's avoid breaking them if we can.

**Review by kushagraThapar — APPROVED** [review]

**Comment by FabianMeiswinkel** [comment]
Both failures (both cases queryDistinctDocuments) are unrelated to my change. @kushagraThapar  or @moderakh can one of you please override the checks to get this merged?

**Pull request merged**